idaes/models_extra/power_generation/unit_models/heat_exchanger_common.py

```diff
@@ -0,0 +1,764 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Methods shared by the CrossFlowHeatExchanger1D and Heater1D models.
+"""
+# Presently the tube methods are not shared. I'm not sure why I chose to extract
+# them here, but I don't want to change things this late into development. --Doug
+
+# Import Pyomo libraries
+from pyomo.environ import (
+    Var,
+    Param,
+    value,
+    units as pyunits,
+)
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+from pyomo.dae import DerivativeVar
+
+# Import IDAES cores
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.constants import Constants as const
+import idaes.core.util.scaling as iscale
+from idaes.core.util.misc import add_object_reference
+
+__author__ = "Jinliang Ma, Douglas Allan"
+
+
+def _make_geometry_common(blk, shell_units):
+    # Number of tube columns in the cross section plane perpendicular to shell side fluid flow (y direction)
+    blk.ncol_tube = Var(
+        initialize=10.0, doc="Number of tube columns", units=pyunits.dimensionless
+    )
+
+    # Number of segments of tube bundles
+    blk.nseg_tube = Var(
```

Comment:
Minor comment: I would generally encourage longer names to avoid any confusion about what these are. Not that important for contrib models however.